cinder/volume/manager.py

```diff
@@ -2566,6 +2566,11 @@ def _report_driver_status(self, context):
             volume_stats = self.driver.get_volume_stats(refresh=True)
             if self.extra_capabilities:
                 volume_stats.update(self.extra_capabilities)
+                if "pools" in volume_stats:
+                    for pool in volume_stats["pools"]:
+                        pool.update(self.extra_capabilities)
+                else:
+                    volume_stats.update(self.extra_capabilities)
```

Comment:
This is required for all pools, otherwise the scheduler never sees the extra_capabilities.  This was a bug that I found while working on this and I got it applied upstream as well.   Some drivers report pools, while others do not.  Either way, if the extra_capabilities are not added to each pool, then the scheduler will never see them.  This is required.